### DISCUSSION TITLE
Hosting a static configuration (nginx.conf) using Nginx on Oracle Cloud Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by berbarroso** [nginx/kubernetes-ingress#3535] [Q&A]
Hi,

How to host a static configuration (nginx.conf) using Nginx on Oracle Cloud Kubernetes Cluster?

I only found one for Docker. 

Here is the link:
https://hub.docker.com/_/nginx/

**Comment by brianehlert** [comment]
Are you just looking for a pre-built nginx container?
You found the official one.

**Reply by berbarroso** [reply]
I already have one. 

I deploy nginx using manifests:
https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/

**Reply by brianehlert** [reply]
NGINX Ingress Controller is a special use case.
It is focused on NGINX as a reverse proxy use cases.
Yes it has enough flexibility you could do this.
But using nginx pods and mounting a common config using a shared volume is generally how we would think about using NGINX as a service.

**Reply by berbarroso** [reply]
Yes, I'm having problem converting the docker commands since I'm using Oracle Cloud Kubernetes Cluster for configuring nginx.conf file. My apologies for not specifying my cloud provider from early on.

**Reply by berbarroso** [reply]
I tried it on Docker and this is my command:
docker run --name webhost-nginx -v C:/Users/GemberB/nginx/webhost-nginx/nginx.conf:/etc/nginx/nginx.conf:ro -d nginx

Syntax:
docker run  --name <NAME OF THE CONTAINER>  -v /host/path/nginx.conf:/etc/nginx/nginx.conf:ro -d nginx

How can I convert it on a kubectl command?

**Reply by brianehlert** [reply]
NGINX ingress controller is not designed to be used in that way.  The configuration is applied based on Ingress or CRD configuraiton objects in the K8s API.

You should simply be deploying a generic nginx container with a shared volume (defined in the deployment spec) and your nginx.conf in the volume.

**Comment by vepatel** [comment]
Closing this discussion as this isn't specific to this project.
